docs/tutorial.md

```diff
@@ -0,0 +1,182 @@
+# Oria Operator Tutorial
+
+An in-depth walkthrough of building and running a oria-operator with memcached-operator.
+
+## Prerequisites
+
+- [Operator SDK](https://sdk.operatorframework.io/docs/installation/) v1.21.0 or newer
+- User authorized with `cluster-admin` permissions so that you can install an operator, CRDs and change rbac.
+- [GNU Make](https://www.gnu.org/software/make/)
+
+## Overview
+
+We will create a sample project to let you know how it works and this sample will:
+
+- Create a `memcached-operator` using below mentioned steps. For more infomation refer to this [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/)
+- Generate the manifests using `config/default`
+- Comment out `ClusterRole` and `ClusterRoleBinding`. 
+- Create a `ClusterRole` template in a `ScopeTemplate` and associate it with a `ScopeInstance`
+- Run the operator, the `memcached-operator` will start complaining about `RBAC`.
+- Run the `oria-operator`, it will create `ClusterRole` and `ClusterRoleBinding`. This will make sure that `memcached-operator` will not complain about `RBAC`. 
+
+### Install and run oria-operator
+
+Clone the [github repo](https://github.com/operator-framework/oria-operator.git) and  run the below commands on the root of the directory.
+
+```
+$ make install
+
+$ make run
+```
+
+### Create a new project
+
+Use the [Operator SDK](https://sdk.operatorframework.io/docs/installation/) CLI to create a new memcached-operator from the [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/).
+
+
+```sh
+mkdir -p $HOME/projects/memcached-operator
+cd $HOME/projects/memcached-operator
+# we'll use a domain of example.com
+# so all API groups will be <group>.example.com
+operator-sdk init --domain example.com --repo github.com/example/memcached-operator
```

Comment:
I would break all of these commands into their own code block for easier copy-paste. 

Alternatively we could combine all of these into one "command" like:
```
mkdir -p $HOME/projects/memcached-operator && \
cd $HOME/projects/memcached-operator && \
operator-sdk init --domain example.com --repo github.com/example/memcached-operator
```

Reply:
I am okay with the above approach but not sure if every user wants to create a directory at `$HOME/projects/memcached-operator`.

Sometimes I create it somewhere else. WDYT @everettraven ?

Reply:
I think we can keep them separate commands if we wanted and just have the docs be like:

Create a directory for the memcached operator:
```
mkdir -p <directory>
```
Navigate into the newly created directory and initialize the project with:
```
operator-sdk init --domain example.com --repo github.com/example/memcached-operator
```

Reply:
yeah, that sounds good to me.

Reply:
updated